### PR TITLE
Adding a filter compoent class and passing on to all serializers in it

### PR DESCRIPTION
**Pull request by findsah**
Filter Integration with ChartSerializer and all other serializers

### CODE REVIEW

File: dashboards/component/chart/serializers.py [hunk 1]
```diff
@@ -5,12 +5,27 @@
...
+import asset_definitions  # Import asset_definitions module if it's part of your project
+
 import pandas as pd
 import plotly.graph_objs as go
 
 from dashboards.meta import ClassWithMeta
+import django_filters
+import random
 
+class FilterComponent(Component):
```
Comment (jdwpeterson) [inline dashboards/component/chart/serializers.py:20]: This shouldn't be defined here.  Needs to go into `dashboards/component/` as its own file - see `form.py` for example.  The file should probably be called `filter.py`

File: dashboards/component/chart/serializers.py [hunk 2]
```diff
@@ -5,11 +5,17 @@
 from django.db.models import Model
 from django.template.loader import render_to_string
 
-import asset_definitions
+from dashboards.component.filter import FilterComponent
+# Import missing components
+from dashboards.meta import ClassWithMeta
```
Comment (jdwpeterson) [inline dashboards/component/chart/serializers.py:10]: imported below - line 16

File: dashboards/component/chart/serializers.py [hunk 3]
```diff
@@ -191,3 +193,31 @@ class Meta(ModelDataMixin.Meta, PlotlyChartSerializer.Meta):
         pass
 
     _meta: Type["ChartSerializer.Meta"]
+
+class StatSerializer(ModelDataMixin, PlotlyChartSerializer, FilterComponent):
```
Comment (jdwpeterson) [inline dashboards/component/chart/serializers.py:197]: no need to add this here.  it should stay in the stat folder

Comment (findsah) [reply dashboards/component/chart/serializers.py:197]: okay I will fix these and push them with new update

File: dashboards/component/chart/serializers.py [hunk 4]
```diff
@@ -180,8 +183,7 @@ class Meta(PlotlyChartSerializerMixin.Meta, BaseChartSerializer.Meta):
     class Media:
         js = ("dashboards/vendor/js/plotly.min.js",)
 
-
-class ChartSerializer(ModelDataMixin, PlotlyChartSerializer):
+class ChartSerializer(ModelDataMixin, PlotlyChartSerializer, FilterComponent):
```
Comment (jdwpeterson) [inline dashboards/component/chart/serializers.py:186]: `FilterComponent` object should be passed in when the Serializer object is called, not inherited from, see https://github.com/wildfish/django-dashboards/blob/main/dashboards/component/base.py#L118

Comment (findsah) [reply dashboards/component/chart/serializers.py:186]: okay

File: demos/dashboard/demo/kitchensink/charts.py [hunk 5]
```diff
@@ -7,6 +7,8 @@
 from dashboards.component.chart import ChartSerializer
 
 
+
```
Comment (jdwpeterson) [inline demos/dashboard/demo/kitchensink/charts.py:10]: extra spaces not needed

File: dashboards/templates/dashboards/components/filter.html [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<form id="filter-form" hx-post="{% url 'your_dashboard_refresh_url' %}" hx-trigger="submit">
```
Comment (jdwpeterson) [inline dashboards/templates/dashboards/components/filter.html:1]: This should not be an example.  It needs to be a working template that knows how to display the django filter form e.g.

```
<form method="get">
        {{ rendered_value.filter.form.as_p }}
        <input type="submit" />
    </form>
    ```
    You will need more than this

File: dashboards/component/stat/serializers.py [hunk 7]
```diff
@@ -41,6 +42,7 @@ class Meta(ClassWithMeta.Meta):
         model: Optional[Model] = None
         title: Optional[str] = ""
         unit: Optional[str] = ""
+        filter_component: Optional[Type[FilterComponent]] = None
```
Comment (jdwpeterson) [inline dashboards/component/stat/serializers.py:45]: don't think this needs to be here - should be passed into serialize instead as an attr

File: dashboards/templates/dashboards/components/filter.html (hunk 6 above)
Comment (findsah) [reply dashboards/templates/dashboards/components/filter.html:1]: Sure, I will create a working template also looking into dashboard. Html code to render it over there

File: dashboards/component/stat/serializers.py [hunk 8]
```diff
@@ -110,6 +119,9 @@ def get_value(self) -> Any:
     @classmethod
     def serialize(cls, **kwargs) -> StatSerializerData:
         self = cls()
+        filters = {}  # You need to obtain filters from the FilterComponent
```
Comment (jdwpeterson) [inline dashboards/component/stat/serializers.py:122]: filter component object should be passed in to serialize instead.  you would then need to do something like 

```
f = ProductFilter(request.GET, queryset=self.get_queryset())
queryset = f.get_queryset()
```

This is not right, you will been to read the django filter docs to find out how to do it correctly

**Pull request closed without merging**